server/server.go

```diff
@@ -43,6 +44,18 @@ func setupRouter(ctx context.Context, logger *logrus.Logger, testRouter bool) (c
 	}
 	extensions := extension.OfferedExtensions
 	r.Mount("/extensions", controller.ExtensionsRouter(extensions, testRouter))
+
+	// Add profiling flag to enable profiling routes.
+	if os.Getenv("PPROF_ENABLED") != "" {
+		// pprof attaches routes to default serve mux
+		// host:6061/debug/pprof/
+		go func() {
+			if err := http.ListenAndServe(":6061", http.DefaultServeMux); err != nil {
+				logger.WithError(err).Error("Server failed to start")
+			}
+		}()
+	}
```

Comment:
`setupRouter` is not an appropriate place for this. One would not expect such a side effect from creating a router.

Please move it to `StartServer`, before or after the metric service is started (before line 67 or after 74).